src/collections/Form/Form.js

```diff
@@ -1,143 +1,264 @@
-    this.refresh()
-  }
+  // warn about form nodes missing a "name"
+  getNodeName = (node) => {
+    const name = node.name
 
-  componentWillUnmount() {
-    _.invoke(this, 'element.off')
-  }
+    if (!name) {
+      const errorMessage = [
+        'Encountered a form control node without a name attribute.',
+        'Each node in a group should have a name.',
+        'Otherwise, the node will serialize as { "undefined": <value> }.',
+      ].join(' ')
+      console.error(errorMessage, node) // eslint-disable-line no-console
+    }
 
-  static _meta = {
-    name: 'Form',
-    type: META.TYPES.COLLECTION,
+    return name
   }
+}
 
-  static Field = FormField
-  static Fields = FormFields
+function formSerializer(formNode) {
+  debug('formSerializer()')
+  const json = {}
+  // handle empty formNode ref
+  if (!formNode) return json
 
-  plugin(...args) {
-    return this.element.form(...args)
-  }
+  // ----------------------------------------
+  // Checkboxes
+  // Single: { name: value|bool        }
+  // Group:  { name: [value|bool, ...] }
 
-  refresh() {
-    this.element = $(this.refs.element)
-    this.element.form(_.pick(this.props, [
-      'keyboardShortcuts',
-      // validation
-      'delay', 'duration', 'fields', 'on', 'inline', 'revalidate', 'transition',
-      // callbacks
-      'onValid', 'onInvalid', 'onSuccess', 'onFailure',
-    ]))
-  }
+  _.each(formNode.querySelectorAll('input[type="checkbox"]'), (node, index, arr) => {
+    const name = getNodeName(node)
+    const checkboxesByName = _.filter(arr, { name })
 
-  serializeJson = () => {
-    const form = this.refs.element
-    const json = {}
-
-    _.each(['input', 'textarea', 'select'], (tag) => {
-      _.each(form.getElementsByTagName(tag), (node, index, arr) => {
-        const name = node.getAttribute('name')
-
-        switch (node.getAttribute('type')) {
-          case 'checkbox':
-            json[name] = { checked: node.checked }
-            break
-
-          case 'radio':
-            json[name] = {
-              value: _.find(arr, { name, checked: true }).value,
-            }
-            break
-
-          default:
-            json[name] = { value: node.value }
-            break
-        }
-      })
-    })
-
-    return json
-  }
+    // single: (value|checked)
+    if (checkboxesByName.length === 1) {
+      json[name] = node.checked && node.value !== 'on' ? node.value : node.checked
+      debugSerializedResult(json, name, node)
+      return
+    }
+
+    // groups (checked): [value, ...]
+    if (!Array.isArray(json[name])) json[name] = []
+    if (node.checked) json[name].push(node.value)
+    debugSerializedResult(json, name, node)
+
+    // in dev, warn about multiple checkboxes with a default browser value of "on"
+    if (process.NODE_ENV !== 'production' && node.value === 'on') {
+      const errorMessage = [
+        "Encountered a checkbox in a group with the default browser value 'on'.",
```

Comment:
This message makes sense because you explained it to me, but I think a user seeing this would be pretty lost. Maybe a link to an issue or a simplified rewrite would help.